### PR TITLE
8306456: Don't leak _worklist's memory in PhaseLive::compute

### PR DESCRIPTION
**Pull request by jdksjolen**
`PhaseLive::compute` used to do this: `_worklist = new (_arena) Block_List();`. This allocates the `Block_List` to the `_arena`, but the backing array is allocated on the resource area: `Block_List() : Block_Array(Thread::current()->resource_area()), _cnt(0) {}`. This causes at most 5 worklists and at least 4 worklists to be created and not freed until the compilation is finished. This patch allocates the worklist within `PhaseLive::compute`:s local resource mark.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306456](https://bugs.openjdk.org/browse/JDK-8306456): Don't leak _worklist's memory in PhaseLive::compute

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [e1191341](https://git.openjdk.org/jdk/pull/13535/files/e119134167b8c6fbceaec52f71b83398bcf2fe85)
 * [Dean Long](https://openjdk.org/census#dlong) (@dean-long - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13535/head:pull/13535` \
`$ git checkout pull/13535`

Update a local copy of the PR: \
`$ git checkout pull/13535` \
`$ git pull https://git.openjdk.org/jdk.git pull/13535/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13535`

View PR using the GUI difftool: \
`$ git pr show -t 13535`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13535.diff">https://git.openjdk.org/jdk/pull/13535.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13535#issuecomment-1514840710)

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

File: src/hotspot/share/opto/live.cpp [hunk 1]
```diff
@@ -88,6 +87,9 @@ void PhaseLive::compute(uint maxlrg) {
 
   _free_IndexSet = nullptr;
 
+  Block_List wl;
+  _worklist = &wl;
+
```
Comment (dean-long) [inline src/hotspot/share/opto/live.cpp:92]: Now `_worklist` is a dangling pointer to released stack memory at the end of this method.  How do we make sure it isn't used?

**Review by dean-long — CHANGES_REQUESTED** [review]

File: src/hotspot/share/opto/live.cpp (hunk 1 above)
Comment (jdksjolen) [reply src/hotspot/share/opto/live.cpp:91]: I removed the `_worklist` local variable and pass it into the functions that need it instead.

**Comment by jdksjolen** [comment]
I also added some style changes, to make the code look more like the rest of Hotspot. I didn't convert the whole file though, only areas that I touched.

**Review by dean-long — APPROVED** [review]
Looks good.

**Pull request closed without merging**